### PR TITLE
feat: highlight changed

### LINKED ISSUE #267: Support highlighting the changed values
Support for highlighting changes when updated, just like the debugger did in vscode.

图片

I'd like to work on this if you agree.

### PR DESCRIPTION
**Pull request by EYHN**
Add a highlighting effect to the key of the changed data.

> Shortcoming: Because the value is only compared within the `<DataKeyPair />` component, highlighting only display at the deepest key where changed. Adding and removing properties do not cause highlighting.

## Add Option

* enableHighlight: Whether to highlight the changed

## Screenshot

Feel free to make your requests and suggestions.

close #267

### CODE REVIEW

File: src/stores/JsonViewerStore.ts [hunk 1]
```diff
@@ -49,6 +51,7 @@ export const createJsonViewerStore = <T = unknown> (props: JsonViewerProps<T>) =
   return create<JsonViewerState>()((set, get) => ({
     // provided by user
     enableClipboard: props.enableClipboard ?? true,
+    enableHighlight: props.enableHighlight ?? true,
```
Comment (pionxzh) [inline src/stores/JsonViewerStore.ts:54]: I think this feature should be disabled by default.
```suggestion
    enableHighlight: props.enableHighlight ?? false,
```

File: src/components/DataKeyPair.tsx [hunk 2]
```diff
@@ -82,6 +84,43 @@ export const DataKeyPair: FC<DataKeyPairProps> = (props) => {
...
+
+      if (typeof value === 'object' || typeof value === 'function') {
+        return false
+      }
+
+      if (value !== prevValue) {
+        return true
+      }
+    }
+    return false
```
Comment (pionxzh) [inline src/components/DataKeyPair.tsx:106]: This part can be extracted to `utils` with some test

**Comment by pionxzh** [comment]
> Adding and removing properties do not cause highlighting.

It's ok. We can enhance this in the future.

File: src/components/DataKeyPair.tsx [hunk 3]
```diff
@@ -82,6 +84,43 @@ export const DataKeyPair: FC<DataKeyPairProps> = (props) => {
...
+  }, [enableHighlight, prevValue, value])
+  const highlightContainer = useRef<HTMLElement>()
+  useEffect(() => {
+    if (highlightContainer.current && isHighlight) {
+      highlightContainer.current.animate(
```
Comment (pionxzh) [inline src/components/DataKeyPair.tsx:111]: The compatability of `animate` API looks ok.
But as the fact that we need to support down to `ES5`, maybe we can do a `?.animate` or `'animatet' in highlighContainer.current`?

File: src/components/DataKeyPair.tsx (hunk 2 above)
Comment (pionxzh) [reply src/components/DataKeyPair.tsx:106]: wait, this is not a normal compare function... nah ignore what I said

Comment (EYHN) [reply src/components/DataKeyPair.tsx]: ok, i added some comments here

**Comment by EYHN** [comment]
All done, please review again 🙏

**Comment by pionxzh** [comment]
LGTM. Thanks for your contribution.

**Review by pionxzh — APPROVED** [review]

**Comment by pionxzh** [comment]
I kind of want to update the base16 color to make it yellow 🤔 maybe I will try in another PR.

**Pull request merged**